pkg/dataplane/item.go

```diff
@@ -85,55 +85,61 @@ func (i Item) GetFieldUint64(name string) (uint64, error) {
 }
 
 // For internal use only - DO NOT USE!
-func (i Item) GetShard() (int, []*ItemChunkData, *ItemChunkMetadata, *ItemCurrentChunkMetadata, error) {
+func (i Item) GetShard() (map[int]*ItemChunk, *ItemCurrentChunkMetadata, error) {
 	const streamDataPrefix = "__data_stream["
 	const streamMetadataPrefix = "__data_stream_metadata["
 	const offsetPrefix = "__data_stream[0000]["
 
-	var chunkDataArray []*ItemChunkData
-	chunkMetaData := ItemChunkMetadata{}
 	currentChunkMetadata := ItemCurrentChunkMetadata{}
-	var chunkID int
+	chunkMap := make(map[int]*ItemChunk)
```

Comment:
if `chunkMap` contains multiple entries in what order (if it matters) should the chunks be restored?

Reply:
I don't commit any spacial order in the map,
this pr not changing the expected restore order.